krnlmon_options.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KRNLMON_OPTIONS_H_
+#define KRNLMON_OPTIONS_H_
+
+#if defined(DPDK_TARGET)
+// DPDK options
+
+#elif defined(ES2K_TARGET)
+// ES2K options
+#define LAG_OPTION 1
+
+#else
+#error "TDI target not specified"
```

Comment:
This would result in a compile-time error for Tofino target correct? Should we change the message that "only DPDK and ES2K targets are supported" instead?

Reply:
This is actually a belt-and-suspenders warning.

Standalone cmake build for `tofino` target:

```text
dfoster@dgfoster-mobl1:~/work/latest/krnlmon/krnlmon$ cmake -B build -DTDI_TARGET=tofino
  .
  .
-- WITH_OVSP4RT="ON"
Building TOFINO_TARGET
CMake Error at cmake/standalone.cmake:86 (message):
  Tofino target not supported
Call Stack (most recent call first):
  CMakeLists.txt:17 (include)

-- Configuring incomplete, errors occurred!
See also "/home/dfoster/work/latest/krnlmon/krnlmon/build/CMakeFiles/CMakeOutput.log".
```

Standalone Bazel build for `tofino` target:

```text
dfoster@dgfoster-mobl1:~/work/latest/krnlmon/krnlmon$ bazel build --config tofino //:krnlmon
Starting local Bazel server and connecting to it...
ERROR: Config value 'tofino' is not defined in any .rc file
```

Standalone Bazel build with no target specified:

```text
dfoster@dgfoster-mobl1:~/work/latest/krnlmon/krnlmon$ bazel build //:krnlmon
ERROR: /home/dfoster/work/latest/krnlmon/krnlmon/switchlink/sai/BUILD.bazel:41:19: configurable attribute "defines" in //switchlink/sai:switchlink_handle_link doesn't match this configuration: must specify --define target={dpdk|es2k}
```

Integrated cmake build for `tofino` target:

```text
dfoster@dgfoster-mobl1:~/work/latest$ ./make-all.sh --target=tofino --no-build
  .
  .
Building TOFINO_TARGET
-- WITH_KRNLMON="OFF"
-- WITH_OVSP4RT="OFF"
  .
  .
-- Configuring done
-- Generating done
-- Build files have been written to: /home/dfoster/work/latest/build

dfoster@dgfoster-mobl1:~/work/latest$ cmake --build build -j4 --target krnlmon
make: *** No rule to make target 'krnlmon'.  Stop.
```

Explanation: the `tofino` target disables krnlmon and ovsp4rt.

Reply:
My usual policy is to validate parameters at the interface and not do so internally. Conditionals make the code more complex. They add visual noise, and they leave the reader uncertain about the program's error states. (They also tend to reflect the _developer's_ uncertainty about the error states!)

I made an exception here because -- with the proliferation of ways to build krnlmon (standalone cmake, integrated cmake, and standalone Bazel), as well as the increasing number of build configurations (ovs={on|off}, lnw_version={2|3}) -- I was no longer certain myself. Maybe having a check in the code itself is called-for, and this is a pretty good place to put one.

You do, nonetheless, raise a valid point. Error messages should be informative, and it costs us nothing to do so here.